react/week4/github-user-searcher/src/Git.js

```diff
@@ -0,0 +1,40 @@
+import React, { useEffect } from "react";
+import { GitUserProvider } from "./gitUserContext";
+import GitUsers from "./GitUsers";
+
+const Git = () => {
+
+    const [search, setsearch] = React.useState('');
+    const [users, setUsers] = React.useState([]);
+    const [error, setError] = React.useState();
+    const [isLoading, setIsLoading] = React.useState(false);
+    const getUsers = async (search) => {
+        if (search !== "") {
+            setIsLoading(true);
+            await fetch(`https://api.github.com/search/users?q=${search}&limit=20`)
+                .then((response) => response.json())
+                .then((data) => {
+                    if (data.message) {
+                        setError(data.message);
+                    }
+                    setUsers(data.items)
+                    setIsLoading(false);
+                })
+        } else {
+            setUsers([]);
+        }
+    }
+
+    useEffect(() => {
+        getUsers(search);
+    }, [search]);
+
+    const githubUserContent = { search, error, users, isLoading }
+    return (
+        <GitUserProvider value={githubUserContent}>
```

Comment:
I'd recommend putting all of this in the Context file you created with the Provider.

Then have all this logic abstracted into there.

Then in that file pass the props children and wrap in there

```
function GitUserProvider({children}){
   ...all of this logic state + useEffect

   const githubUserContent = { search, error, users, isLoading } 

   return <GitUserProvider value={githubUserContent}> {children} </GitUserProvider>
}
```